### PR TITLE
Use kubevirtci to deploy local Kubernetes cluster, add basic e2e tests

### PR DESCRIPTION
**Pull request by phoracek**
This PR introduces helper commands to use kubevirtci local Kubernetes cluster.

It also introduces few basic functional tests.

### CODE REVIEW

File: test.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+from kubernetes import client, config
```
Comment (phoracek) [inline test.py]: commited by mistake, will be removed

File: tests/_cli.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+def run(node, command):
+    p = subprocess.Popen(
+        ['docker', 'exec', 'kubevirt-' + node, 'ssh.sh'] + command,
+        stdin=subprocess.PIPE,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+    )
+    stdout, stderr = p.communicate()
+
+    return p.returncode, stdout.split('\n')[:-1], stderr.split('\n')[2:-1]
```
Comment (EdDev) [inline tests/_cli.py]: It will be better to raise and add all the info needed there.
When all is working as expected, you do not need to handle any asserts on the success code.

File: cluster/build.sh [hunk 3]
```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/bash -ex
+#!/usr/bin/bash -e
```
Comment (EdDev) [inline cluster/build.sh]: How are these changes relevant? (same for sync.sh)

File: cluster/up.sh [hunk 4]
```diff
@@ -3,4 +3,10 @@
 source ./cluster/helpers.sh
 
 $gocli run --random-ports --nodes 2 --background kubevirtci/k8s-1.10.3
+
+# refresh with new port numbers
```
Comment (EdDev) [inline cluster/up.sh]: Still not clear why you do it.

File: tests/_kubecli.py [hunk 5]
```diff
@@ -0,0 +1,7 @@
+import urllib3
+
+from kubernetes import config
+
+urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
+
+config.load_kube_config('./kubeconfig')
```
Comment (EdDev) [inline tests/_kubecli.py]: It is nicer (and safer) to place it under a (session) fixture that uses 'autouse'.

File: tests/test_kubernetes.py [hunk 6]
```diff
@@ -1,8 +1,21 @@
+from kubernetes import client
+
 from . import _kubectl
+from . import _kubecli  # NOQA: initialize kubernetes client
+
+import urllib3
+urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
```
Comment (EdDev) [inline tests/test_kubernetes.py]: Nicer if it will be a fixture with autouse.

**Comment by EdDev** [comment]
On other projects, the tests modules use *_test naming format. I know that pytest picks them up automatically, not sure about the ones that start with test_* as you set here.

I find it easier to locate a test when the module/subject it tests is in front, but that is a matter of taste.

**Comment by phoracek** [comment]
@EdDev thanks a lot, will post fixes soon.

File: cluster/build.sh [hunk 7]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash -e
```
Comment (yuvalif) [inline cluster/build.sh:1]: how is this different than the current "docker-build" target we have? isn't there an overlap?

File: cluster/sync.sh [hunk 8]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash -e
```
Comment (yuvalif) [inline cluster/sync.sh:1]: shouldn't this be merged with the existing "docker-push" target?

**Comment by yuvalif** [comment]
the current test does not seem to cover any of the plugins. seems like they are just testing the cluster.

**Comment by phoracek** [comment]
@yuvalif I will add tests for bridge plugin in a following patch, tests in this PR are just to test the testing :)

File: cluster/build.sh (hunk 7 above)
Comment (phoracek) [reply cluster/build.sh:1]: I wanted to keep it in Makefile, but I did not figure out how to `source` and obtain the port. Any idea how to fix it?

File: cluster/sync.sh (hunk 8 above)
Comment (yuvalif) [reply cluster/sync.sh:1]: ok. this is only called from ```make cluster-sync```

File: cluster/build.sh (hunk 7 above)
Comment (yuvalif) [reply cluster/build.sh:1]: no need. now is see its is only called from: ```make cluster-sync```

**Review by yuvalif — APPROVED** [review]

**Comment by phoracek** [comment]
Thanks, merging.
